### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by AnicetFantomas**
In this pull request we performed :

- Run a program using the command line.
- Set up associations between classes and objects.

### CODE REVIEW

**Review by Davidosky007 — COMMENTED** [review]
Hi @AnicetFantomas @YannickZahinda 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has the wrong gitflow, kindly create a dev branch and make it your default branch then open a new PR from your feature branch into the dev branch 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: .github/classroom.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    def initialize(label)
+      @label = label
+      @students = []
+    end
+  
+    def add_students(student)
+      @students.push(student)
+      student.classroom = self
+    end
+  end
```
Comment (hiromataba) [inline .github/classroom.rb]: - [ ] This class should not be in the `.github` directory please make it in the root of your project like other classes

**Review by hiromataba — CHANGES_REQUESTED** [review]
Hi @AnicetFantomas     :wave: :wave: 

almost-there

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Your PR has a proper title and description  :+1: 
- Your linters are passing :heavy_check_mark:  
- Profesional README file :rocket: 

## Required Changes ♻️

- [ ] When setting the classroom for a student it also adds it to the classrooms' students, your student class should be like this 👇 

```ruby
attr_reader :classroom

  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 2]
```diff
@@ -2,14 +2,15 @@
...
+  attr_reader :id, :rentals
 
   def initialize(age, name = 'unknown', parent_permision: true)
     super()
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permision = parent_permision
+    @rentals = []
   end
```
Comment (julie-ify) [inline person.rb]: - [ ] I got the attached errors when I tried creating a student. It happens that there is a typographical error in your `Nameable`. Here you have it as `Nameable` while the class itself is `Namable`. Kindly fix this so the error by making use of the correct name of your class.

- [ ] I also got an error about the file name. You have `require './nameable'` but actually the name of your nameable file is `namable`. you should require the file with the correct file name `require './namable'` to fix the error.
chrome_zOwO52dCVV

slack_C7TqCeQGLv

**Review by andersonlebon — COMMENTED** [review]
## Hi @AnicetFantomas 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not match with the project requirements

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @AnicetFantomas and @YannickZahinda , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @AnicetFantomas,

## APPROVED 🟢

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**